source/connectors/kafka-sink-postprocessors.txt

```diff
@@ -9,10 +9,821 @@ Kafka Sink Connector Post-Processors
+
+   * - newName
+     - Contains the replacement text for all matches of the regular expression
+       defined in the ``pattern`` field.
+
+Example
+
+.. code-block:: properties
+
+   field.renamer.mapping=[{"regexp":"^key\\.date.*$","pattern":"_","replace":"-"},{"regexp":"^value\\.crepes\\..*","pattern":"purchased","replace":"quantity"}]
+
+The record contains the following data after applying the ``RenameByMapping``
+post processor:
+
+**Key document**
+
+.. code-block:: json
+   :emphasize-lines: 3,4
+
+   {
+     "city": "Provence",
+     "date-month": "October",
+     "date-day": 17
+   }
+
+**Value document**
+
+.. code-block:: json
+   :emphasize-lines: 3
+
+   {
+     "crepes": {
+       "quantity": 598,
+       "size": "large"
+     }
+   }
+
+The post processor applied the following changes:
+
+* All field names in the key document of the sink record that started with
+  "date" are matched. In the matched fields, all instances of "_" are
+  replaced with "-".
+* All field names in the value document of the sink record that are
+  subdocuments of ``crepes`` are matched. In the matched fields, all
+  instances of "purchased" are replaced with "quantity".
+
+.. admonition:: Ensure renaming does not result in duplicate keys in the same document
+
+   The renaming post processors update the key fields of a JSON document
+   which can result in duplicate keys within a document. They skip the
+   renaming step if the replacement key already exists at the current level.
+
+.. _custom-write-models:
+
+Custom Write Models
+-------------------
+
+A **write model** defines the behavior of bulk write operations made on a
+MongoDB collection. The default write model for the connector is
+:java-docs-latest:`ReplaceOneModel
+<javadoc/com/mongodb/client/model/ReplaceOneModel.html>` with
+:java-docs-latest:`ReplaceOptions </javadoc/com/mongodb/client/model/ReplaceOptions.html>`
+set to upsert mode.
+
+You can override the default write model by specifying a custom one in the
+``mongodb.writemodel.strategy`` configuration setting. The following
+strategies are provided with the connector:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Write Model
+     - Description
+
+   * - ReplaceOneDefaultStrategy
+     - | Replaces at most one document that matches the current document by the ``_id`` field.
+       | Default value for ``writemodel.strategy`` configuration setting.
+
+   * - ReplaceOneBusinessKeyStrategy
+     - | Replaces at most one document that matches filters provided by the ``document.id.strategy`` setting.
+       | Set the following configuration: ``writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.ReplaceOneBusinessKeyStrategy``
+
+       .. seealso:: Example of usage in :ref:`writemodel-strategy-business-key`.
+
+   * - DeleteOneDefaultStrategy
+     - | Deletes at most one document that contain null value sink records that matches the id specified by the ``document.id.strategy`` setting.
```

Comment:
s/matches/match ?

Reply:
I think this is a confusing sentence, so I will rearrange the phrasing to more closely match the other items. Since "matches" is referring to the "at most one document" part, I think the singular form is appropriate. However, the phrase inserted to describe the null value record makes it more confusing what the verb references.